app/actions/remote/channel.ts

```diff
@@ -798,7 +798,7 @@ export async function createDirectChannel(serverUrl: string, userId: string, dis
             const config = await getConfig(database);
             const teammateDisplayNameSetting = getTeammateNameDisplaySetting(preferences || [], config.LockTeammateNameDisplay, config.TeammateNameDisplay, license);
             const {directChannels, users} = await fetchMissingDirectChannelsInfo(serverUrl, [created], currentUser.locale, teammateDisplayNameSetting, currentUser.id, true);
-            created.display_name = directChannels?.[0].display_name || created.display_name;
+            created.display_name = (directChannels?.length && directChannels?.[0].display_name) || created.display_name;
```

Comment:
Why this change? What did you find?

Reply:
It's possible to have `directChannels` be an empty array which would crash here. It's possible that this can't happen through normal usage of the app but in mocked environments like the unit tests it was happening. So I updated this to be resilient to that